src/CMakeLists.txt

```
@@ -1,13 +1,7 @@
-cmake_minimum_required(VERSION 3.0)
-
-# The library
 add_library(example complex.cpp)
 target_include_directories(example PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
+# Setting this to the library only avoids errors in 3rd party dependencies which are compile with e.g. -Werror
+target_compile_options(example PUBLIC -Wall -pedantic)
 # Include code-coverage settings:
 target_link_libraries(example PUBLIC coverage_config)
 
```

While pushing more modern CMake, how about using [target_compile_features](https://cmake.org/cmake/help/latest/command/target_compile_features.html) on the library? Since [CMake 3.8](https://cmake.org/cmake/help/v3.8/prop_gbl/CMAKE_CXX_KNOWN_FEATURES.html) C++ versions are available.
```suggestion
if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.8)
  target_compile_features(example PUBLIC cxx_std_11)
endif()
```

---

I was thinking about going that far but restricted myself on removing the `if` only. Furthermore those compile features have drawbacks too and I didn't want to start a discussion about that. (Using `target_compile_features` could lead to parts of your targets build with differing standard versions which may or may not be a problem for your use case)